accepted/2025/nuget-mcp.md

```diff
@@ -0,0 +1,234 @@
+
+Current list: `npm`, `docker`, `pypi`, `homebrew`
+
+#### Add NuGet to the list
+
+We will work with the MCP server registry team to add `nuget` to the list.
+
+We will publish/link to guidance on how to publish a NuGet-based MCP server to the MCP registry. This step is performed after the package is published to NuGet.org.
+
+### Add support for NuGet packages in VS Code
+
+#### Today's experience
+
+Similar to the previous section, NuGet is not a recognized MCP server host.
+
+![VS Code MCP server installation](../../meta/resources/nuget-mcp/vscode-today.png)
+
+#### Add NuGet to the list
+
+We will work with the VS Code team to add a NuGet option to the list, which updates the user's `mcp.json` to invoke `dnx` with the provided package ID.
+
+Once Visual Studio has a corresponding experience ([the current experience is manually editing the `mcp.json`](https://learn.microsoft.com/en-us/visualstudio/ide/mcp-servers?view=vs-2022)), we will ensure NuGet is supported similarly.
+
+## Rollout plan
+
+1. Enable `McpServer` support on NuGet.org search and the package details page, behind a feature flag.
+2. Work with Microsoft and community MCP server implementers to package as NuGet.
+   - As of June 2025, there are 28 .NET tools on NuGet.org that use the MCP SDK. Newer versions should have the `McpServer` package type.
+   - Some Microsoft MCP servers are implemented in .NET but distributed via npm, for example [@azure/mcp](https://www.npmjs.com/package/@azure/mcp).
+3. Wait for `dnx` to land in a .NET 10 preview.
+4. Work with the MCP registry and VS Code to get NuGet support, perhaps providing PRs/OSS contributions ourselves.
+5. Enable MCP server UI on NuGet.org.
+6. Publish a document on learn.microsoft.com on how to create your own NuGet MCP server. 
+
+## Future Possibilities
+
+We will wait to publish an MCP server template until the .NET MCP SDK has announced a stable API surface area. It is currently in prerelease.
+
+## Prior Art
+
+We should replicate what is already working for npm, PyPI, and Docker.
+
+## Unresolved Questions
+
+- What guidance should be provided for private MCP server implementations?
+  - For example, if you publish an `McpServer` to your Azure DevOps feed, what MCP registry should be used?
+- What schema should be used for a machine-readable MCP server startup instruction?
+  - The package author could include an `mcp.json` in the NuGet package [matching the MCP registry OpenAPI spec](https://github.com/modelcontextprotocol/registry/blob/a4cefcf05f81466ad65e7c3971e76d0f6d60783e/docs/openapi.yaml#L183-L201).
+- How are client runtime requirements expressed?
+  - For example, if an MCP server needs a certain .NET version, how is this communicated to the end user before failure occurs?
```

Comment:
Super interesting question here. Didn't think about this as I was under the impression this would be a .NET SDK version/compatibility issue.